code/game/machinery/barsigns.dm

```diff
@@ -1,133 +1,178 @@
-/obj/structure/sign/barsign // All Signs are 64 by 32 pixels, they take two tiles
+/obj/machinery/barsign // All Signs are 64 by 32 pixels, they take two tiles
 	name = "bar sign"
 	desc = "A bar sign which has not been initialized, somehow. Complain at a coder!"
 	icon = 'icons/obj/barsigns.dmi'
 	icon_state = "empty"
 	req_access = list(ACCESS_BAR)
-	max_integrity = 500
+	obj_integrity = 500
 	integrity_failure = 250
 	armor = list("melee" = 20, "bullet" = 20, "laser" = 20, "energy" = 100, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50, "stamina" = 0)
-	buildable_sign = 0
-
-	var/panel_open = FALSE
+	use_power = ACTIVE_POWER_USE
+	active_power_usage = 150
+	idle_power_usage = 15
+	//Selected barsign being used
 	var/datum/barsign/chosen_sign
 
-/obj/structure/sign/barsign/Initialize(mapload)
+MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/barsign, 32)
+
+/obj/machinery/barsign/Initialize(mapload)
 	. = ..()
 	set_sign(new /datum/barsign/hiddensigns/signoff)
 
-/obj/structure/sign/barsign/proc/set_sign(datum/barsign/sign)
+/obj/machinery/barsign/proc/set_sign(datum/barsign/sign)
 	if(!istype(sign))
 		return
 
-	icon_state = sign.icon
+	if(sign.rename_area)
+		rename_area(src, sign.name)
+
+	chosen_sign = sign
+	update_appearance()
+
+/obj/machinery/barsign/update_icon_state()
+	if(!(machine_stat & (NOPOWER|BROKEN)) && chosen_sign && chosen_sign.icon)
+		icon_state = chosen_sign.icon
+	else
+		icon_state = "empty"
+
+	return ..()
+
+/obj/machinery/barsign/power_change()
+    . = ..()
+    update_icon_state()
+
+/obj/machinery/barsign/update_desc()
+	. = ..()
+
+	if(chosen_sign && chosen_sign.desc)
+		desc = chosen_sign.desc
 
-	if(sign.name)
-		name = "[initial(name)] ([sign.name])"
+/obj/machinery/barsign/update_name()
+	. = ..()
+	if(chosen_sign && chosen_sign.rename_area)
+		name = "[initial(name)] ([chosen_sign.name])"
 	else
 		name = "[initial(name)]"
 
-	if(sign.desc)
-		desc = sign.desc
+/obj/machinery/barsign/update_overlays()
+	. = ..()
 
-	if(sign.rename_area && sign.name)
-		rename_area(src, sign.name)
+	if(!(machine_stat & (NOPOWER|BROKEN)))
+		return
 
-	return sign
+	if(chosen_sign && chosen_sign.light_mask)
+		. += emissive_appearance(icon, "[chosen_sign.icon]-light-mask", src)
 
-/obj/structure/sign/barsign/proc/set_sign_by_name(sign_name)
-	for(var/d in subtypesof(/datum/barsign))
-		var/datum/barsign/D = d
-		if(initial(D.name) == sign_name)
-			var/new_sign = new D
-			return set_sign(new_sign)
+/obj/machinery/barsign/update_appearance(updates=ALL)
+	. = ..()
+	if(!(machine_stat & (NOPOWER|BROKEN)))
+		set_light(0)
+		return
+	if(chosen_sign && chosen_sign.neon_color)
+		set_light(MINIMUM_USEFUL_LIGHT_RANGE, 0.7, chosen_sign.neon_color)
```

Comment:
This is always going to return with `set_light(0)` as `machine_stat` is going to be 0.